Sources/Datadog/Logging/Log/LogEventEncoder.swift

```diff
@@ -18,24 +19,45 @@ internal struct LogEvent: Encodable {
         case critical
         case emergency
     }
+    public struct Attributes {
+        /// Log custom attributes, They are subject for sanitization.
+        public var userAttributes: [String: Encodable]
+        /// Log attributes added internally by the SDK. They are not a subject for sanitization.
+        internal let internalAttributes: [String: Encodable]?
+    }
 
-    let date: Date
-    let status: Status
-    let message: String
-    let error: DDError?
-    let serviceName: String
-    let environment: String
-    let loggerName: String
-    let loggerVersion: String
-    let threadName: String?
-    let applicationVersion: String
-    let userInfo: UserInfo
-    let networkConnectionInfo: NetworkConnectionInfo?
-    let mobileCarrierInfo: CarrierInfo?
-    var attributes: LogAttributes
-    var tags: [String]?
-
-    func encode(to encoder: Encoder) throws {
+    /// The log's timestamp
+    public let date: Date
+    /// The log status
+    public let status: Status
+    /// The log message
+    public var message: String
+    /// The associated log error
+    internal let error: DDError?
```

Comment:
There are two motivations behind the scrubbing API, uniformly met in Tracing and RUM scrubbing:
* 1/ the user can see all user-controlled values that will be sent to Datadog,
* 2/ the user can redact some of these values (we decide what is mutable and what is not).

Here, this internal `error` does clearly hide some information (1), because [it is serialized](https://github.com/DataDog/dd-sdk-ios/blob/181359bd655b894041702332c26fa05b2d451990/Sources/Datadog/Logging/Log/LogEncoder.swift#L109-L114) into 3 distinct fields:
```swift
// Encode log.error properties
if let someError = log.error {
    try container.encode(someError.type, forKey: .errorKind)
    try container.encode(someError.message, forKey: .errorMessage)
    try container.encode(someError.stack, forKey: .errorStack)
}
```
How about defining these 3 values as explicit properties on `LogEvent`:
```swift
public let error: LogEvent.Error?

public struct LogEvent.Error {
   public let kind: String?
   public let message: String?
   public let stack: String?
}
```
and extracting their values from `DDError` before creating the `LogEvent` (so in `LogEventBuilder`)? Then, we can decide (2) if any of these values should be mutable `var` - in RUM we allow redaction of [`error.message`](https://github.com/DataDog/dd-sdk-ios/blob/181359bd655b894041702332c26fa05b2d451990/Sources/Datadog/RUM/DataModels/RUMDataModels.swift#L1115) and [`error.stack`](https://github.com/DataDog/dd-sdk-ios/blob/181359bd655b894041702332c26fa05b2d451990/Sources/Datadog/RUM/DataModels/RUMDataModels.swift#L1124) but not [`error.type`](https://github.com/DataDog/dd-sdk-ios/blob/181359bd655b894041702332c26fa05b2d451990/Sources/Datadog/RUM/DataModels/RUMDataModels.swift#L1127). WDYT?